rest/route/commit_queue.go

```diff
@@ -77,24 +86,76 @@ func (cq *commitQueueDeleteItemHandler) Parse(ctx context.Context, r *http.Reque
 }
 
 func (cq *commitQueueDeleteItemHandler) Run(ctx context.Context) gimlet.Responder {
-	found, err := cq.sc.CommitQueueRemoveItem(cq.project, cq.item)
+	itemRemoved, err := cq.sc.CommitQueueRemoveItem(cq.project, cq.item)
 	if err != nil {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "can't delete item"))
 	}
-	if !found {
+	if itemRemoved == nil {
 		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
 			StatusCode: http.StatusNotFound,
 			Message:    fmt.Sprintf("no matching item for project ID '%s', item '%s'", cq.project, cq.item),
 		})
 	}
 
+	// Send GitHub status
+	projectRef, err := cq.sc.FindProjectById(cq.project)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "can't find project %s", cq.project))
+	}
+	if projectRef.CommitQueue.PatchType == commitqueue.PRPatchType {
+		pr, err := cq.sendDeletedStatusToPR(ctx, projectRef.Owner, projectRef.Repo, itemRemoved.Issue)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "can't send status for PR on '%s/%s' #%s", projectRef.Owner, projectRef.Repo, itemRemoved.Issue))
+		}
+		projectConfig, err := cq.sc.GetProjectConfigforPR(ctx, cq.settings, projectRef, pr)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "can't get config for PR '%s/%s' #%s", projectRef.Owner, projectRef.Repo, itemRemoved.Issue))
+		}
+		for _, module := range itemRemoved.Modules {
+			projectModule, err := projectConfig.GetModuleByName(module.Module)
+			if err != nil {
+				return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "can't get module for module name '%s'", module.Module))
+			}
+			owner, repo, err := thirdparty.ParseGitUrl(projectModule.Repo)
+			if err != nil {
+				return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "can't parse module repo '%s' for owner/repo", projectModule.Repo))
+			}
+			_, err = cq.sendDeletedStatusToPR(ctx, owner, repo, module.Issue)
```

Comment:
Should we just send it to the main repository? One notification might be sufficient.

Reply:
done.